src/API/WesternStatesWater.WestDaat.Client.Functions/WaterAllocation.cs

```diff
@@ -113,97 +113,97 @@ public async Task<IActionResult> GetWaterRightsEnvelope([HttpTrigger(Authorizati
         {
             var result = await _waterAllocationManager.GetSiteDigest(siteUuid);
 
-            return new OkObjectResult(result);
+            return await JsonResult(request, result);
         }
 
-        [FunctionName(nameof(GetSiteLocation)), AllowAnonymous]
-        public async Task<IActionResult> GetSiteLocation([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/SiteLocation")] HttpRequest request, string siteUuid)
+        [Function(nameof(GetSiteLocation))]
+        public async Task<HttpResponseData> GetSiteLocation([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/SiteLocation")] HttpRequestData request, string siteUuid)
         {
             var result = await _waterAllocationManager.GetWaterSiteLocation(siteUuid);
 
-            return new OkObjectResult(JsonSerializer.Serialize(result));
+            return await JsonResult(request, result);
         }
 
-        [FunctionName(nameof(GetWaterSiteSourceListByUuid)), AllowAnonymous]
-        public async Task<IActionResult> GetWaterSiteSourceListByUuid([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/Sources")] HttpRequest request, string siteUuid)
+        [Function(nameof(GetWaterSiteSourceListByUuid))]
+        public async Task<HttpResponseData> GetWaterSiteSourceListByUuid([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/Sources")] HttpRequestData request, string siteUuid)
         {
             var result = await _waterAllocationManager.GetWaterSiteSourceInfoListByUuid(siteUuid);
 
-            return new OkObjectResult(result);
+            return await JsonResult(request, result);
         }
 
-        [FunctionName(nameof(GetWaterSiteRightsListByUuid)), AllowAnonymous]
-        public async Task<IActionResult> GetWaterSiteRightsListByUuid([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/Rights")] HttpRequest request, string siteUuid)
+        [Function(nameof(GetWaterSiteRightsListByUuid))]
+        public async Task<HttpResponseData> GetWaterSiteRightsListByUuid([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "Sites/{siteUuid}/Rights")] HttpRequestData request, string siteUuid)
         {
             var result = await _waterAllocationManager.GetWaterSiteRightsInfoListByUuid(siteUuid);
 
-            return new OkObjectResult(result);
+            return await JsonResult(request, result);
         }
 
 
-        [FunctionName(nameof(DownloadWaterRights)), AllowAnonymous]
-        public async Task<EmptyResult> DownloadWaterRights([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "WaterRights/download")] HttpRequest request)
+        [Function(nameof(DownloadWaterRights))]
+        public async Task<HttpResponseData> DownloadWaterRights([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "WaterRights/download")] HttpRequestData request)
```

Comment:
We need to update this to return `Task<IActionResult>`. Isolated doesn't support chunking and will cause the a memory issue as it will load the whole file in memory. We included the nuget package already so we need to update this return type.

https://github.com/Azure/azure-functions-host/issues/4926#issuecomment-1675192574

Reply:
Good to resolve?